GLMakie/assets/shader/volume.frag

```diff
@@ -96,32 +105,33 @@ vec3 gennormal(vec3 uvw, float d)
         return vec3(0, 0, -1);
     }
 
-    a.x = texture(volumedata, uvw - vec3(d,0.0,0.0)).r;
-    b.x = texture(volumedata, uvw + vec3(d,0.0,0.0)).r;
+    a.x = texture(volumedata, uvw - vec3(o.x, 0.0, 0.0)).r;
+    b.x = texture(volumedata, uvw + vec3(o.x, 0.0, 0.0)).r;
 
-    a.y = texture(volumedata, uvw - vec3(0.0,d,0.0)).r;
-    b.y = texture(volumedata, uvw + vec3(0.0,d,0.0)).r;
+    a.y = texture(volumedata, uvw - vec3(0.0, o.y, 0.0)).r;
+    b.y = texture(volumedata, uvw + vec3(0.0, o.y, 0.0)).r;
 
-    a.z = texture(volumedata, uvw - vec3(0.0,0.0,d)).r;
-    b.z = texture(volumedata, uvw + vec3(0.0,0.0,d)).r;
-    return normalize(a-b);
+    a.z = texture(volumedata, uvw - vec3(0.0, 0.0, o.z)).r;
+    b.z = texture(volumedata, uvw + vec3(0.0, 0.0, o.z)).r;
+
+    vec3 diff = a - b;
+    float n = length(diff);
+
+    if (n < 0.000000000001) // 1e-12
```

Comment:
```suggestion
    if (n < 1e-12)
```

Reply:
I don't think GLSL supports that syntax?

Reply:
you're right https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)